app/controllers/application_controller.rb

```diff
@@ -2,4 +2,21 @@
 
 class ApplicationController < ActionController::Base
   include Clearance::Controller
+
+  before_action :set_time_zone
+
+  def set_time_zone
+    Time.zone = browser_time_zone
+  end
+
+  private
+
+  def browser_time_zone
+    browser_tz = ActiveSupport::TimeZone.find_tzinfo(cookies[:timezone])
+    ActiveSupport::TimeZone
+      .all
+      .find { |zone| zone.tzinfo == browser_tz } || Time.zone
+  rescue TZInfo::UnknownTimezone, TZInfo::InvalidTimezoneIdentifier
+    Time.zone
+  end
```

Comment:
Is putting `rescue` here essentially making the function call is act as `begin`?

Reply:
Yep, that's right. A method definition itself can act as the `begin`.

You can do this with blocks too

```ruby
array.map do |item|
 # do some stuff
rescue 
  puts "rescued"
end

```